clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -156,14 +155,31 @@ public synchronized int requestUpdateForNewTopics() {
     }
 
     /**
-     * Request an update for the partition metadata iff the given leader epoch is newer than the last seen leader epoch
+     * Request an update for the partition metadata iff we encounter a leader epoch that is newer than the last seen leader epoch
```

Comment:
It might be useful to mention here that this method is reserved for cases where the leader epoch is derived from some external source (such as committed offsets or fetched records). Maybe also document the returned value.